### PR TITLE
[CELEBORN-1644] Optimize handle merged data on stage end

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
1. Add worker readable address to commit files response logs, to avoid redundant logs
2. Pass stage end status to the worker to avoid unnecessary retry 

### Why are the changes needed?
Committing files will take a period of time if a speculation task is pushing merged data to the worker. This will cause the client to retry until the committing files is complete or client timeout. This PR intend to optimize the logic to avoid unnecessary retry.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
GA.

### CODE REVIEW

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java [hunk 1]
```diff
@@ -1391,10 +1391,16 @@ public void onSuccess(ByteBuffer response) {
                 groupedBatchId,
                 Arrays.toString(batchIds));
             pushState.removeBatch(groupedBatchId, hostPort);
-            if (response.remaining() > 0 && response.get() == StatusCode.MAP_ENDED.getValue()) {
-              mapperEndMap
-                  .computeIfAbsent(shuffleId, (id) -> ConcurrentHashMap.newKeySet())
-                  .add(mapId);
+            if (response.remaining() > 0) {
+              int retCode = response.get();
+              if (retCode == StatusCode.MAP_ENDED.getValue()) {
```
Comment (SteNicholas) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:1396]: Is there any compatibility problem for old client version and new server version?

Comment (FMX) [reply client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:1396]: This is possible. I'll fix it.

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala [hunk 2]
```diff
@@ -490,7 +492,14 @@ class PushDataHandler(val workerSource: WorkerSource) extends BaseMessageHandler
               ByteBuffer.wrap(Array[Byte](StatusCode.HARD_SPLIT.getValue)))
           }
         } else {
-          if (storageManager.shuffleKeySet().contains(shuffleKey)) {
+          // This means that this stage is ended and invoked commit files by stage end
+          if (presumptiveEndedShuffles.contains(shuffleKey)) {
```
Comment (RexXiong) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala:496]: Seems we can check stage_end before checking map_end?

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java [hunk 3]
```diff
@@ -1391,10 +1391,16 @@ public void onSuccess(ByteBuffer response) {
...
+                mapperEndMap
+                    .computeIfAbsent(shuffleId, (id) -> ConcurrentHashMap.newKeySet())
+                    .add(mapId);
+              }
+              if (retCode == StatusCode.STAGE_ENDED.getValue()) {
```
Comment (RexXiong) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:1401]: elseif maybe better

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala [hunk 4]
```diff
@@ -490,6 +498,7 @@ class PushDataHandler(val workerSource: WorkerSource) extends BaseMessageHandler
               ByteBuffer.wrap(Array[Byte](StatusCode.HARD_SPLIT.getValue)))
           }
         } else {
+          // This means that this stage is ended and invoked commit files by stage end
```
Comment (RexXiong) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala:501]: should remove this line~

File: client/src/main/scala/org/apache/celeborn/client/commit/CommitHandler.scala [hunk 5]
```diff
@@ -439,7 +442,7 @@ abstract class CommitHandler(
         replicaIds)
     }
 
-    doParallelCommitFiles(shuffleId, shuffleCommittedInfo, params, commitFilesFailedWorkers)
+    doParallelCommitFiles(shuffleId, shuffleCommittedInfo, params, commitFilesFailedWorkers, true)
```
Comment (RexXiong) [inline client/src/main/scala/org/apache/celeborn/client/commit/CommitHandler.scala:445]: Only the ReducePartitionCommitHandler can set stageEnd to true when call handleFinalCommitFiles method.

**Pull request closed without merging**